### PR TITLE
Add items2dict example

### PR DESCRIPTION
**Pull request by Andersson007**
Add items2dict example

Relates to https://github.com/ansible-community/community-examples/issues/5

### CODE REVIEW

File: tests/integration/targets/core/tasks/items2dict_1.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# As input we have the list [{'key': 'string1', 'value': 'string2'}].
+# We want to transform the list into a dictionary that have 'string1'
+# as a key and string2 as its value.
+# The following variable represents the list data before
+# applying the items2dict filter plugin:
+- name: Set the input list
+  set_fact:
+   input_list:
+     - key: string1
+       value: string2
```
Comment (jillr) [inline tests/integration/targets/core/tasks/items2dict_1.yml:14]: Generally +1 all of this, but I think it's worth showing (either here or in a second example) a list of multiple dicts to make it more obvious.  So like:

```
  - name: Set the input list
    set_fact:
      input_list:
      - key: string1
        value: string2
      - key: string3
        value: string4
  # After applying the items2dict filter plugin, we expect to
  # get a dictionary of the desired form:
  - name: Transform list to dictionary using items2dict plugin
    set_fact:
      result_dict: '{{ input_list | items2dict(key_name="key", value_name="value") }}'

  # Let's check if the result variable contains the data we want:
  - name: Check the result
    assert:
      that:
        - result_dict['string1'] == 'string2'
        - result_dict['string3'] == 'string4'
```

Comment (Andersson007) [reply tests/integration/targets/core/tasks/items2dict_1.yml:14]: The idea sounds good, changed the example, thanks! PTAL

**Comment by Andersson007** [comment]
@jillr @samccann PTAL

**Review by jillr — APPROVED** [review]
lgtm, thanks @Andersson007

**Comment by Andersson007** [comment]
@jillr thanks for reviewing!
I'll re-evaluate things and get back with new questions / suggestions in the TODO issue or in the community topic and we'll continue the discussion there.

**Pull request merged**